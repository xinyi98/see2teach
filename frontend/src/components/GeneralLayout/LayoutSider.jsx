import {
  DesktopOutlined,
  DashboardOutlined,
  CalendarOutlined,
  PlusOutlined,
} from '@ant-design/icons';
import { Button, Typography } from 'antd';
import { Link, useLocation } from 'react-router-dom';
import { Layout, Menu, Image } from 'antd';
import { Mobile } from '../../utils/media-breakpoints';
import UserDropdown from './UserDropdown';
import { Tooltip } from 'antd';

const { Sider } = Layout;
const { Title } = Typography;

/**
 * This is the sider of the layout. Contains all links to pages and title and allows it to be collapsed.
 * @returns JSX of sider layout.
 */
const LayoutSider = () => {
  const location = useLocation();

  return (
    <Sider
      className="sidebar"
      theme="light"
      breakpoint="lg"
      collapsedWidth={0}
      style={{ position: 'sticky', top: 0, left: 0, height: '100vh' }}
    >
      <div // Place second child (user dropdown) at the bottom of the sidebar on mobile
        style={{
          display: 'flex',
          flexDirection: 'column',
          height: '100%',
          justifyContent: 'space-between',
        }}
      >
        <div>
          <Link to="/">
            <div className="logo">
              <Image width={35} src="/uoa.svg" preview={false} alt="uoa logo" />
              <Title level={3} style={{ float: 'right', marginRight: 10 }}>
                See2Teach
              </Title>
            </div>
          </Link>

          <Mobile>
            <Link to="/request-review">
              <Button style={{ marginLeft: 20, marginBottom: 12 }} icon={<PlusOutlined />}>
                Request a Review
              </Button>
            </Link>
          </Mobile>

          <Menu theme="light" mode="inline" selectedKeys={[location.pathname]}>
            <Menu.Item key="/" icon={<DashboardOutlined />}>
              <Link to="/">Dashboard</Link>
            </Menu.Item>
            <Menu.Item key="/my-reviews" icon={<DesktopOutlined />}>
              <Tooltip
                mouseEnterDelay={0.5}
                placement="right"
                title="Others’ observations of my courses."
              >
                <Link to="/my-reviews">My Reviews</Link>
              </Tooltip>
            </Menu.Item>
            <Menu.Item key="/requested-reviews" icon={<CalendarOutlined />}>
              <Tooltip
                mouseEnterDelay={0.5}
                placement="right"
                title="My observations of others’ courses"
              >
                <Link to="/requested-reviews">Requested Reviews</Link>
              </Tooltip>
            </Menu.Item>
          </Menu>
        </div>

        <Mobile>
          <div style={{ margin: 12 }}>
            <UserDropdown />
          </div>
        </Mobile>
      </div>
    </Sider>
  );
};

export default LayoutSider;
