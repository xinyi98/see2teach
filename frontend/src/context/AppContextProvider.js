import React, { useContext } from 'react';
import { createContext, useState } from 'react';

export const AppContext = createContext();

/**
 * Use this hook to use the user context in the application
 * @returns Context containing {user, setUser}
 */
export function useAppContext() {
  return useContext(AppContext);
}

/**
 * This provider contains the user information. Wrap around APp.
 * @param {*} param0 Children that will be placed between provider
 * @returns JSX of the provider
 */
export function AppContextProvider({ children }) {
  const [user, setUser] = useState({
    id: window.sessionStorage.getItem('user_id'),
    name: window.sessionStorage.getItem('name'),
    upi: window.sessionStorage.getItem('upi'),
  });

  return <AppContext.Provider value={{ user, setUser }}>{children}</AppContext.Provider>;
}
