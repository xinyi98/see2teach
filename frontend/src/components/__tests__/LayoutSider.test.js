import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { AppContextProvider } from '../../context/AppContextProvider';
import LayoutSider from '../GeneralLayout/LayoutSider';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

test('snapshot test for sidebar', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <AppContextProvider>
          <LayoutSider />
        </AppContextProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
