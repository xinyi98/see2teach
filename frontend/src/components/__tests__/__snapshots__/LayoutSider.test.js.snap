// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for sidebar 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-light sidebar"
  style={
    Object {
      "flex": "0 0 200px",
      "height": "100vh",
      "left": 0,
      "maxWidth": "200px",
      "minWidth": "200px",
      "position": "sticky",
      "top": 0,
      "width": "200px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "justifyContent": "space-between",
        }
      }
    >
      <div>
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="logo"
          >
            <div
              className="ant-image"
              style={
                Object {
                  "height": undefined,
                  "width": 35,
                }
              }
            >
              <img
                alt="uoa logo"
                className="ant-image-img"
                onError={[Function]}
                onLoad={[Function]}
                src="/uoa.svg"
                style={
                  Object {
                    "height": undefined,
                  }
                }
              />
            </div>
            <h3
              className="ant-typography"
              style={
                Object {
                  "WebkitLineClamp": undefined,
                  "float": "right",
                  "marginRight": 10,
                }
              }
            >
              See2Teach
            </h3>
          </div>
        </a>
        <ul
          className="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
          data-menu-list={true}
          onKeyDown={[Function]}
          role="menu"
          tabIndex={0}
        >
          <li
            className="ant-menu-item ant-menu-item-selected"
            data-menu-id="rc-menu-uuid-test-/"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="dashboard"
              className="anticon anticon-dashboard ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="dashboard"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M924.8 385.6a446.7 446.7 0 00-96-142.4 446.7 446.7 0 00-142.4-96C631.1 123.8 572.5 112 512 112s-119.1 11.8-174.4 35.2a446.7 446.7 0 00-142.4 96 446.7 446.7 0 00-96 142.4C75.8 440.9 64 499.5 64 560c0 132.7 58.3 257.7 159.9 343.1l1.7 1.4c5.8 4.8 13.1 7.5 20.6 7.5h531.7c7.5 0 14.8-2.7 20.6-7.5l1.7-1.4C901.7 817.7 960 692.7 960 560c0-60.5-11.9-119.1-35.2-174.4zM761.4 836H262.6A371.12 371.12 0 01140 560c0-99.4 38.7-192.8 109-263 70.3-70.3 163.7-109 263-109 99.4 0 192.8 38.7 263 109 70.3 70.3 109 163.7 109 263 0 105.6-44.5 205.5-122.6 276zM623.5 421.5a8.03 8.03 0 00-11.3 0L527.7 506c-18.7-5-39.4-.2-54.1 14.5a55.95 55.95 0 000 79.2 55.95 55.95 0 0079.2 0 55.87 55.87 0 0014.5-54.1l84.5-84.5c3.1-3.1 3.1-8.2 0-11.3l-28.3-28.3zM490 320h44c4.4 0 8-3.6 8-8v-80c0-4.4-3.6-8-8-8h-44c-4.4 0-8 3.6-8 8v80c0 4.4 3.6 8 8 8zm260 218v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8zm12.7-197.2l-31.1-31.1a8.03 8.03 0 00-11.3 0l-56.6 56.6a8.03 8.03 0 000 11.3l31.1 31.1c3.1 3.1 8.2 3.1 11.3 0l56.6-56.6c3.1-3.1 3.1-8.2 0-11.3zm-458.6-31.1a8.03 8.03 0 00-11.3 0l-31.1 31.1a8.03 8.03 0 000 11.3l56.6 56.6c3.1 3.1 8.2 3.1 11.3 0l31.1-31.1c3.1-3.1 3.1-8.2 0-11.3l-56.6-56.6zM262 530h-80c-4.4 0-8 3.6-8 8v44c0 4.4 3.6 8 8 8h80c4.4 0 8-3.6 8-8v-44c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Dashboard
              </a>
            </span>
          </li>
          <li
            className="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/my-reviews"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="desktop"
              className="anticon anticon-desktop ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="desktop"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              <a
                href="/my-reviews"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                My Reviews
              </a>
            </span>
          </li>
          <li
            className="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/requested-reviews"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 24,
              }
            }
            tabIndex={-1}
          >
            <span
              aria-label="calendar"
              className="anticon anticon-calendar ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              <a
                href="/requested-reviews"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                Requested Reviews
              </a>
            </span>
          </li>
        </ul>
        <div
          aria-hidden={true}
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</aside>
`;
